internal/pkg/runtime/engines/singularity/prepare_linux.go

```diff
@@ -96,14 +97,31 @@ func (e *EngineOperations) prepareUserCaps() error {
 			sylog.Debugf("Ignoring group %d: %s", g, err)
 			continue
 		}
-		authorizedCaps, _ := capConfig.CheckGroupCaps(gr.Name, caps)
+		authorizedCaps, unauthorizedCaps := capConfig.CheckGroupCaps(gr.Name, caps)
 		if len(authorizedCaps) > 0 {
 			sylog.Debugf("%s group capabilities %s added", gr.Name, strings.Join(authorizedCaps, ","))
 			commonCaps = append(commonCaps, authorizedCaps...)
 		}
+		if len(unauthorizedCaps) > 0 {
+			commonUnauthorizedCaps = append(commonUnauthorizedCaps, unauthorizedCaps...)
+		}
 	}
 
 	commonCaps = capabilities.RemoveDuplicated(commonCaps)
+	commonUnauthorizedCaps = capabilities.RemoveDuplicated(commonUnauthorizedCaps)
+
+	// remove authorized capabilities from unauthorized capabilities list
+	// to end with the really unauthorized capabilities
+	for _, cap := range commonCaps {
```

Comment:
"cap" is the built-in function to obtain a slice's capacity. Better use something else just to avoid confusion.

Reply:
will fix